mlrun/api/schemas/feature_store.py

```diff
@@ -1,45 +1,42 @@
 from datetime import datetime
 from typing import Optional, List
 
-from pydantic import BaseModel
+from pydantic import BaseModel, Extra
+from .object import ObjectMetadata
 
 
 class Feature(BaseModel):
     name: str
     value_type: str
     labels: Optional[dict]
 
+    class Config:
+        extra = Extra.allow
+
 
 class Entity(BaseModel):
     name: str
     value_type: str
     labels: Optional[dict]
 
-
-class FeatureSetMetadata(BaseModel):
-    name: str
-    tag: Optional[str]
-    labels: Optional[dict]
-    updated: Optional[datetime]
-    uid: Optional[str]
+    class Config:
+        extra = Extra.allow
 
 
 class FeatureSetSpec(BaseModel):
     entities: List[Entity]
     features: List[Feature]
 
+    class Config:
+        extra = Extra.allow
+
 
 class FeatureSet(BaseModel):
-    metadata: FeatureSetMetadata
+    metadata: ObjectMetadata
     spec: FeatureSetSpec
-    status: Optional[dict]
-
```

Comment:
No status anymore ?

Reply:
No. Status was there just as a placeholder for a dictionary stored as JSON. Since I now added the `full_object` JSON object, then there is no need to treat `status` independently - the only exception is the `state` field which is extracted from the full object if provided and stored in the DB table.

Reply:
I see, IMO the `FeatureSet` schema should not allow extra fields, i.e. only metadata spec and status, the underlying objects should allow extra.
WDYT ?

Reply:
Talked offline, decided to go with what I suggested

Reply:
Done. Note that extra fields are allowed in the schema (it's not that strict), but they will not be serialized/de-serialized, so will not be kept in DB. Only  spec/metadata/status will be kept as first-level attributes of FeatureSet.